internal/transport/keepalive_test.go

```diff
@@ -44,6 +44,7 @@ import (
 )
 
 const defaultTestTimeout = 10 * time.Second
+const defaultTestShortTimeout = 10 * time.Millisecond
```

Comment:
Shouldn't we have a similar test to verify transport doesn't close until keepalive goroutine finish?

Reply:
We will not test that and leave that upto the leak checker becuase determining and controlling when `keepalive` function closes is difficult with changing the actual function 
(to add a testhook or something like that).

Reply:
I see. okay